rustls/src/msgs/handshake.rs

```diff
@@ -1287,9 +1287,18 @@ impl<'a> Deref for CertificateChain<'a> {
 }
 
 impl TlsListElement for CertificateDer<'_> {
-    const SIZE_LEN: ListLength = ListLength::U24 { max: 0x1_0000 };
+    const SIZE_LEN: ListLength = ListLength::U24 {
+        max: CERTIFICATE_MAX_SIZE_LIMIT,
+    };
 }
 
+/// TLS has a 16MB size limit on any handshake message,
+/// plus a 16MB limit on any given certificate.
+///
+/// We contract that to 64KB to limit the amount of memory allocation
+/// that is directly controllable by the peer.
+pub(crate) const CERTIFICATE_MAX_SIZE_LIMIT: usize = 0x1_0000;
+
 // TLS1.3 changes the Certificate payload encoding.
 // That's annoying. It means the parsing is not
 // context-free any more.
```

Comment:
This is a carry-over from `main` that isn't related to the PR but I'm curious if this comment should be moved somewhere else. It seems a bit oprhaned hanging out here between `CERTIFICATE_MAX_SIZE_LIMIT` and the `CertificateExtension` enum.

Reply:
Will remove this comment, not really sure what it's adding that is not now obvious from the code.